webui/react/src/pages/ExperimentDetails/CodeViewer/CodeViewer.tsx

```diff
@@ -75,8 +75,8 @@ const sortTree = (a: TreeNode, b: TreeNode) => {
   if (!a.isLeaf && !b.isLeaf) return titleA.localeCompare(titleB) - titleB.localeCompare(titleA);
 
   // had to use RegEx due to some files being ".<filename>"
-  const [stringA, extensionA] = titleA.split(/(?<=[a-zA-Z])\./);
-  const [stringB, extensionB] = titleB.split(/(?<=[a-zA-Z])\./);
+  const [stringA, extensionA] = titleA.split(/^(?=[a-zA-Z])\./);
+  const [stringB, extensionB] = titleB.split(/^(?=filename)\./);
```

Comment:
i think `filename` is a typo? what do you think about just changing from here to the bottom of this function to the following?
```
  const _a = titleA.split('.').reverse().join('.')
  const _b = titleB.split('.').reverse().join('.')
  return _a.localeCompare(_b)
```

the desired behavior is that we sort by extension, then by filename. 

you can check if it has the desired behavior [here](https://replit.com/@trentwatt/CuddlyGlamorousMigration#index.ts)

Reply:
The code you provide looks good to me with plane directory. I haven't tried it with nested directories, so I'll merge this one first to display `Code` tab on Safari, but this would be nice to change.